src/main/java/com/thebois/models/world/World.java

```diff
@@ -74,7 +76,7 @@ public World(final int worldSize, final Colony colony) {
     }
 
     private void initColony() {
-        colony = new Colony(findEmptyPositions(pawnCount));
+        colony = new Colony(findEmptyPositions(pawnCount), this);
```

Comment:
Feels like this creates some sort of circular dependency. In world we have a constructor that can take colony as an parameter but colony now obviously have a constructor that have world as parameter. Seems like a code smell to me

Reply:
Note that it's implicitly casting `World` into `IFinder`, so it's not a double dependency at least!